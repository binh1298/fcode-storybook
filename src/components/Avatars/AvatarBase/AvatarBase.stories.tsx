import AvatarBase, { AvatarBaseProps } from ".";

import { Story } from "@storybook/react";

export default {
    title: "Components/AvatarBase",
    component: AvatarBase,
};

const Template: Story<AvatarBaseProps> = (args) => <AvatarBase {...args} />;

const source = "https://res.cloudinary.com/dq7l8216n/image/upload/v1620235303/FCode-Avatar.png";

export const ImageAvatar = Template.bind({});
ImageAvatar.args = {
    alt: "Fcode avatar",
    src: source,
};

export const ImageAvatarSmall = Template.bind({});
ImageAvatarSmall.args = {
    size: "small",
    alt: "Fcode avatar",
    src: source,
};

export const ImageAvatarLarge = Template.bind({});
ImageAvatarLarge.args = {
    size: "large",
    alt: "Fcode avatar",
    src: source,
};
