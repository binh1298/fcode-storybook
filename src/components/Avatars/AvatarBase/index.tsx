import {
    Avatar as MaterialAvatar,
    AvatarProps as MaterialAvatarProps,
    Box as MaterialBox,
} from "@material-ui/core";

export interface AvatarBaseProps extends MaterialAvatarProps {}

const avatarStyle = {
    width: "inherit",
    height: "inherit",
};

const AvatarBase = (props: AvatarBaseProps) => {
    return (
        <MaterialBox
            width={36}
            height={36}
            borderRadius="50%"
            borderColor="primary.light"
            border={1}
        >
            <MaterialAvatar {...props} style={avatarStyle} />
        </MaterialBox>
    );
};

export default AvatarBase;
