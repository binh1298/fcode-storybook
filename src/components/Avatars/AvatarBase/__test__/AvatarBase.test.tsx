import AvatarBase, { AvatarBaseProps } from "../";

import { render, RenderResult, screen } from "@testing-library/react";

describe("<AvatarBase />", () => {
    let wrapper: RenderResult;

    const props: AvatarBaseProps = {
        children: "Test",
    };

    beforeEach(() => {
        wrapper = render(<AvatarBase {...props} />);
    });

    it("should exist children", () => {
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
