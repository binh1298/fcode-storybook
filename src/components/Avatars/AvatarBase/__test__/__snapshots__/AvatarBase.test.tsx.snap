// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarBase /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      style="width: inherit; height: inherit;"
    >
      Test
    </div>
  </div>
</div>
`;
